src/utilities/time.ts

```diff
@@ -47,23 +210,151 @@ function parseDurationString(durationString: string): ParsedDurationString | nev
     };
   }
 
-  const fullMicrosecondsRegex = new RegExp(`^${validDurationValueRegex}$`);
+  const durationTime = parseDoyOrYmdTime(durationString) as ParsedDurationString;
+  if (durationTime) {
+    return durationTime;
+  }
+
+  matches = new RegExp(`^(?<sign>([+-]?))(?<int>(\\d+))(\\.?<decimal>(\\d+))?$`).exec(durationString);
```

Comment:
I don't think you can have anything before the group name within the group. You'll have to change `(\\.?<decimal>(\\d+))` to `(?<decimal>\\.(\\d+))`.

Reply:
I see. I fixed it now so it will capture the optional decimal 

``^(?<sign>([+-]?))(?<int>(\\d+))((\\.(?<decimal>(\\d+)))?$)``

Reply:
Nice. One small thing left with this is you're not including the decimal in the string, which would result in there not being a difference between `.1` and `.01` when parsed to a number.